stix/operations.py

```diff
@@ -220,43 +266,46 @@ def extract_indicators(config, params):
         file_id = params.get("file_id")
         spec_version = str(config.get("spec_version"))
         try:
-            from integrations.crudhub import download_file_from_cyops
-            res = download_file_from_cyops(file_id)
-            file_path = join('/tmp', res['cyops_file_path'])
+            if 'api/3/' in file_id:
+                from integrations.crudhub import download_file_from_cyops
+                res = download_file_from_cyops(file_id)
+                file_path = join('/tmp', res['cyops_file_path'])
+            else:
+                file_path = join('/tmp', file_id)
             with open(file_path) as attachment:
                 json_data = attachment.read()
         except:
             json_data = _make_request(file_id, "get")
         data = json.loads(json_data)
         for ioc in data["objects"]:
-            if ioc[
-                "type"] == "indicator" and spec_version == "2.0" and "spec_version" not in ioc and "certificate" not in \
-                    ioc["pattern"]:
-                indicators.append(stix_spec_version(ioc, spec_version))
-            elif ioc["type"] == "indicator" and spec_version == "2.1" and ioc[
-                "spec_version"] == "2.1" and "certificate" not in ioc["pattern"]:
-                indicators.append(stix_spec_version(ioc, spec_version))
+            if ioc["type"] == "indicator" and "spec_version" in ioc.keys() and spec_version == "2.1":
+                indicators.append(stix_spec(ioc, spec_version, params))
+            elif ioc["type"] == "indicator" and "spec_version" not in ioc.keys() and spec_version == "2.0":
+                indicators.append(stix_spec(ioc, spec_version, params))
         if len(indicators) > 0:
             return indicators
         else:
-            raise ConnectorError("Not valid STIX specification file")
+            raise ConnectorError(
+                "Either Indicators not found in STIX file or Connector Configuration STIX Spec Version is not valid ")
```

Comment:
Either the Input file is empty or the specification version for its content is not supported